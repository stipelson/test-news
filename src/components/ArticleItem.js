import React from 'react';
import PropTypes from 'prop-types';
/* import Row from 'emerald-ui/lib/Row';
import Col from 'emerald-ui/lib/Col'; */

const ArticleItem = ({ title, body, image, url }) => {
  return (
    <section>
      <a
        className="list-item"
        href={url}
        target="_blank"
        rel="noopener noreferrer"
      >
        <img
          src={image}
          alt={title || 'Article image'}
          className="list-item-image"
        />
        <h3>{title}</h3>
        <p>{body}</p>
      </a>
    </section>
  );
};

ArticleItem.propTypes = {
  title: PropTypes.string,
  body: PropTypes.string,
  image: PropTypes.string,
  url: PropTypes.string,
};

export default ArticleItem;
