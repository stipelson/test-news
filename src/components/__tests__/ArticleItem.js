import React from 'react';
import renderer from 'react-test-renderer';

import ArticleItem from '../ArticleItem';

describe('ArticleItem: render', () => {
  it('renders article item correctly', () => {
    const tree = renderer.create(<ArticleItem />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('has the link class', () => {
    const component = renderer.create(<ArticleItem />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    let instance = component.root;

    const link = instance.findByType('a');
    expect(link.props.className).toEqual('list-item');
  });

  it('has the image class', () => {
    const component = renderer.create(<ArticleItem />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    let instance = component.root;

    const img = instance.findByType('img');
    expect(img.props.className).toEqual('list-item-image');
  });

  it('Image ever has alt', () => {
    const component = renderer.create(<ArticleItem />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    let instance = component.root;

    const img = instance.findByType('img');
    expect(img.props.alt).toEqual('Article image');
  });

  it('Image has alt equal to title', () => {
    const title = 'Test Article Title';
    const component = renderer.create(<ArticleItem title={title} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    let instance = component.root;

    const img = instance.findByType('img');
    expect(img.props.alt).toEqual(title);
  });

  it('Title prop is rendered correctly', () => {
    const title = 'Test Article Title';
    const component = renderer.create(<ArticleItem title={title} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    let instance = component.root;

    const titleNode = instance.findByType('h3');
    expect(titleNode.children).toEqual([title]);
  });

  it('Body prop is rendered correctly', () => {
    const body = 'Test Article Title';
    const component = renderer.create(<ArticleItem body={body} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    let instance = component.root;

    const bodyNode = instance.findByType('p');
    expect(bodyNode.children).toEqual([body]);
  });
});
