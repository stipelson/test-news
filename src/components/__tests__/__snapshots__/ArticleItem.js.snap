// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleItem: render Body prop is rendered correctly 1`] = `
<section>
  <a
    className="list-item"
    rel="noopener noreferrer"
    target="_blank"
  >
    <img
      alt="Article image"
      className="list-item-image"
    />
    <h3 />
    <p>
      Test Article Title
    </p>
  </a>
</section>
`;

exports[`ArticleItem: render Image ever has alt 1`] = `
<section>
  <a
    className="list-item"
    rel="noopener noreferrer"
    target="_blank"
  >
    <img
      alt="Article image"
      className="list-item-image"
    />
    <h3 />
    <p />
  </a>
</section>
`;

exports[`ArticleItem: render Image has alt equal to title 1`] = `
<section>
  <a
    className="list-item"
    rel="noopener noreferrer"
    target="_blank"
  >
    <img
      alt="Test Article Title"
      className="list-item-image"
    />
    <h3>
      Test Article Title
    </h3>
    <p />
  </a>
</section>
`;

exports[`ArticleItem: render Title prop is rendered correctly 1`] = `
<section>
  <a
    className="list-item"
    rel="noopener noreferrer"
    target="_blank"
  >
    <img
      alt="Test Article Title"
      className="list-item-image"
    />
    <h3>
      Test Article Title
    </h3>
    <p />
  </a>
</section>
`;

exports[`ArticleItem: render has the image class 1`] = `
<section>
  <a
    className="list-item"
    rel="noopener noreferrer"
    target="_blank"
  >
    <img
      alt="Article image"
      className="list-item-image"
    />
    <h3 />
    <p />
  </a>
</section>
`;

exports[`ArticleItem: render has the link class 1`] = `
<section>
  <a
    className="list-item"
    rel="noopener noreferrer"
    target="_blank"
  >
    <img
      alt="Article image"
      className="list-item-image"
    />
    <h3 />
    <p />
  </a>
</section>
`;

exports[`ArticleItem: render renders article item correctly 1`] = `
<section>
  <a
    className="list-item"
    rel="noopener noreferrer"
    target="_blank"
  >
    <img
      alt="Article image"
      className="list-item-image"
    />
    <h3 />
    <p />
  </a>
</section>
`;
